File: 14. Longest Common Prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.size() == 0)
            return "";
        string res = "";
        for (int j = 0; j < strs[0].size(); j++)
        {
            for (int i = 1; i<strs.size(); i++)
            {
                if (j >= strs[i].size() || strs[i][j] != strs[0][j])
                    return res;
            }
            res += strs[0][j];
        }
        return res;
    }
};

File: 14. Longest Common Prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if len(strs) == 0:
            return ""
        elif len(strs) == 1:
            return strs[0]
        res = ""
        ind = 0
        while True:
            temp = ""
            for str in strs:
                if ind == len(str):
                    return res
                if temp == "":
                    temp = str[ind]
                if str[ind] != temp:
                    return res
            res += temp
            ind += 1
